packages/tspd/src/gen-extern-signatures/gen-extern-signatures.ts

```diff
@@ -84,7 +121,7 @@ export async function generateExternSignatureForExports(
   } catch (e) {}
   await host.mkdirp(outDir);
 
-  const files = await generateExternDecorators(program, pkgJson.name, {
+  const files = await generateExternDecorators(program, pkgJson.name, decoratorLocations, {
```

Comment:
Here it seems like we're still only compiling the main export (aside we may want to be smarter than just assuming the first one is always the main export, but that's an existing quirk). While we're going through and compiling N Programs (one per export path) for the purpose of collecting the decorator locations, here we just compile the program once via the main export, which may not include all the decorators defined in all the subpaths.